### PR TITLE
Extend doc section on particle filter workflows

### PR DESCRIPTION
**Pull request by sbastrakov**
I think this was rather unclear for users: #3666 , #3774 , #3780 . This PR at least clarifies the common use case of needing particle global position.

And may help do other `particle + some custom data` cases by example. Of course, there an issue is that in case there is no pre-existing wrapper, a user has to do it themselves. Which is mostly a copy-paste of existing wrappers, but not very intuitive.

### CODE REVIEW

File: docs/source/usage/workflows/particleFilters.rst [hunk 1]
```diff
@@ -60,10 +60,108 @@ and add ``ParticlesForwardPinhole`` to the ``AllParticleFilters`` list:
...
 .. code:: cpp
 
-   using AllParticleFilters = MakeSeq_t<
-       All,
-       ParticlesForwardPinhole
-   >;
+       using AllParticleFilters = MakeSeq_t<
+           All,
+           ParticlesForwardPinhole
+       >;
```
Comment (sbastrakov) [inline docs/source/usage/workflows/particleFilters.rst:66]: Note: these were misaligned in the existing example, fixed as well.

File: docs/source/usage/workflows/particleFilters.rst [hunk 2]
```diff
@@ -60,10 +60,108 @@ and add ``ParticlesForwardPinhole`` to the ``AllParticleFilters`` list:
...
+   } // namespace picongpu
+
+Filtering Based on Global Position
+""""""""""""""""""""""""""""""""""
+
+A particle only stores its relative position inside the cell.
+Thus, with a simple filter like one shown above, it is not possible to get global position of a particle.
+However, there are helper wrapper filters that provide such information in addition to the particle data.
+
+For a special case of slicing along one axis there is a simple existing filter that only needs to be parametrized:
```
Comment (PrometheusPi) [inline docs/source/usage/workflows/particleFilters.rst:79]: only a suggestion:
```suggestion
For a special case of slicing along one axis there is a simple existing filter that only needs to be parametrized:
```
```suggestion
For a special case of slicing along one axis this is a simple existing filter that only needs to be parametrized:
```

File: docs/source/usage/workflows/particleFilters.rst [hunk 3]
```diff
@@ -60,10 +60,108 @@ and add ``ParticlesForwardPinhole`` to the ``AllParticleFilters`` list:
...
+           {
+               static constexpr char const* name = "areaFilter";
+
+               template<typename T_Particle>
+               HDINLINE bool operator()(
+                   DataSpace<simDim> const totalCellOffset,
+                   T_Particle const & particle
+               )
+               {
+                   /* Here totalCellOffset is cell index of the particle in the total coordinate system.
```
Comment (PrometheusPi) [inline docs/source/usage/workflows/particleFilters.rst:146]:
```suggestion
                   /* Here totalCellOffset is cell index of the particle in the total coordinate system.
```
```suggestion
                   /* Here totalCellOffset is the cell index of the particle in the total coordinate system.
```

**Review by PrometheusPi — DISMISSED** [review]

File: docs/source/usage/workflows/particleFilters.rst (hunk 3 above)
Comment (PrometheusPi) [reply docs/source/usage/workflows/particleFilters.rst:146]: I do not know, why the diff appears twice (and once without any changes - seems to be an issue with GitHub - I deleted the comment, but the same issue appeared again.)

**Comment by sbastrakov** [comment]
Thanks @PrometheusPi , commited your suggestions. Used the bew suggestion batch thingy on github, hopefully it doesn't break anything.

**Review by PrometheusPi — APPROVED** [review]

**Comment by sbastrakov** [comment]
@psychocoderHPC do you know why CI is stuck in this PR? Maybe cuz the no-compile badge is not supported after the recent changes?

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**